packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/README.md

```diff
@@ -0,0 +1,87 @@
+- `compatibleRuntimes`: runtimes that are compatible with the provider's source code
+- `handler`: the name of the method within your code that the provider calls to execute your function
+
+The [config](./config.ts) file is structured with the top most level mapping to a `aws-cdk-lib` module, i.e., `aws-s3`, `aws-dynamodb`, etc. Each service can contain one or more provider modules. Provider modules are containers for custom resource providers and will be rendered as a code generated file. Each provider module can contain one or more `ProviderProps` objects. Each `ProviderProps` object contains all the necessary information required to generate a single custom resource provider. The following example shows a more structural breakdown of how the [config](./config.ts) file is configured:
+
+```ts
+const config = {
+  'aws-s3': { // the aws-cdk-lib module
+    'replica-provider': [ // the custom resource provider module
+      // custom resource provider defined as ProviderProps object
+      {
+        // the custom resource provider type - `CdkFunction`
+        type: ComponentType.CDK_FUNCTION,
+        // the source code that the provider will execute
+        sourceCode: path.resolve(__dirname, '..', 'aws-dynamodb', 'replica-handler', 'index.ts'),
+        // runtimes that are compatible with the source code
+        compatibleRuntimes: [Runtime.NODEJS_18_X],
+        // the handler in the source code that the provider will execute
+        handler: 'index.onEventHandler',
+      },
+    ],
+  },
+  'aws-stepfunctions-tasks': {
+    // contains multiple custom resource provider modules
+    'eval-nodejs-provider': [
+      {
+        // the custom resource provider type - `CdkSingletonFunction`
+        type: ProviderType.CDK_SINGLETON_FUNCTION,
+        sourceCode: path.resolve(__dirname, '..', 'aws-stepfunctions-tasks', 'eval-nodejs-handler', 'index.ts'),
+        compatibleRuntimes: [Runtime.NODEJS_18_X],
+      },
+    ],
+    'role-policy-provider': [
+      {
+        type: ProviderType.CDK_SINGLETON_FUNCTION,
+        sourceCode: path.resolve(__dirname, '..', 'aws-stepfunctions-tasks', 'role-policy-handler', 'index.py'),
+        compatibleRuntimes: [Runtime.PYTHON_3_9],
+        // prevent minify and bundle flag is set since the source code is a python file
+        preventMinifyAndBundle: true,
+      },
+    ],
+  },
+};
+```
+
+Code generation for the provider modules is triggered when this package - `@aws-cdk/custom-resource-handlers` - is built. Importantly, this framework is also responsible for minifying and bundling the custom resource providers' source code and dependencies. A flag named `preventMinifyAndBundle` can be configured as part of the `ProviderProps` to prevent minifying and bundling the source code for a specific provider. This flag is `false` by default and is only needed for Python files or for JavaScript/TypeScript files containing require imports.
```

Comment:
nit: Swap this around to be the positive: `minifyAndBundle: true` is the default and it can be set to `false`.